src/Model/File.php

```diff
@@ -108,7 +108,7 @@ protected function init(): void
         $this->onHookShort(Model::HOOK_AFTER_DELETE, function () {
             $path = $this->get('location');
             if ($path && $this->flysystem->fileExists($path)) {
-                $files = (clone $this->getModel())->addCondition(
+                $files = (new self($this->getModel()->getPersistence()))->addCondition(
```

Comment:
this is not ok as model can be extended/modified

Reply:
Hi @mvorisek Thank you for your note, but do not understand what you mean: we need to check unconditioned if other tokens exist with the same location. How do we then accomplish this?

Reply:
related with https://github.com/atk4/data/pull/1054